### PR TITLE
#143 - Add code coverage support

### LINKED ISSUE #143: Add code coverage support
## Background
Add code coverage support to be able measure code quality.

## Feature
New ability to measure current code coverage as one of QA metrics.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- added new plugin dependency
- extended readme.md file with how to get code coverage report
- extended build logic to add JaCoCo non-default settings

Closes #143

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -426,3 +426,16 @@ on business requirements. This table represents all currently supported measurem
...
+```sbt
+sbt jacoco 
+```
+For example modules:
+```sbt
+sbt examples/jacoco s3sdkExamples/jacoco
+```
+Code coverage will be generated on path:
+```
+{local-path}\atum\{module}\target\scala-2.XY\jacoco\report\html
```
Comment (Zejnilovic) [inline README.md:440]:
```suggestion
{project-root}/{module}/target/scala-2.XY/jacoco/report/html
```

File: build.sbt [hunk 2]
```diff
@@ -62,7 +85,18 @@ lazy val core = (project in file("atum"))
...
+  .settings(jacocoCoreReportSettings)
+  .dependsOn(model)
+
+lazy val jacocos3SDKExtReportSettings = Seq(
+  jacocoReportSettings := JacocoReportSettings(
+    s"atum:s3sdkExtension Jacoco Report",
+    None,
+    JacocoThresholds(),
+    Seq(JacocoReportFormats.HTML, JacocoReportFormats.XML),
+    "utf-8")
```
Comment (dk1844) [inline build.sbt:98]: I am no expert in SBT, but I would assume that common `JacocoReportSettings` objects could defined for all usages
```
lazy val commonJacocoReportSettings := JacocoReportSettings(
  formats = Seq(JacocoReportFormats.HTML, JacocoReportFormats.XML) // other options are default
)
```

and then reused
```scala
commonJacocoReportSettings.withTitle("atum:s3sdkExtension Jacoco Report")
```

Looking at the [source code](https://github.com/sbt/sbt-jacoco/blob/main/src/main/scala/com/github/sbt/jacoco/report/JacocoReportSettings.scala), this should be possible. 

This should be, obviously, applicable elsewhere

**Review by dk1844 — COMMENTED** [review]
Otherwise looks good

File: README.md (hunk 1 above)
Comment (miroslavpojer) [reply README.md]: Resolved by commit 5789506e16af458426c5de6e9b110a2315408a2c.

File: build.sbt (hunk 2 above)
Comment (miroslavpojer) [reply build.sbt]: Resolved by commit 5789506e16af458426c5de6e9b110a2315408a2c.

File: build.sbt [hunk 3]
```diff
@@ -149,7 +124,10 @@ lazy val s3sdkExamples = (project in file("examples-s3-sdk-extension"))
     (Compile / compile) := ((Compile / compile) dependsOn printSparkScalaVersion).value, // printSparkScalaVersion is run with compile
     mergeStrategy,
   )
-  .settings(jacocoS3sdkExamplesReportSettings: _*)
+  .settings(
+    jacocoReportSettings := commonJacocoReportSettings.withTitle("atum:s3sdkExamples Jacoco Report"),
+    jacocoExcludes := commonJacocoExcludes
+  )
```
Comment (Zejnilovic) [inline build.sbt:130]: Examples should not have a Jacoco report. We are even skipping tests in them. They are not a deliverable package but rather a showcase.

File: build.sbt [hunk 4]
```diff
@@ -129,17 +109,12 @@ lazy val examples = (project in file("examples"))
     (Compile / compile) := ((Compile / compile) dependsOn printSparkScalaVersion).value, // printSparkScalaVersion is run with compile
     mergeStrategy
   )
+  .settings(
+    jacocoReportSettings := commonJacocoReportSettings.withTitle("atum:examples Jacoco Report"),
+    jacocoExcludes := commonJacocoExcludes
+  )
```
Comment (Zejnilovic) [inline build.sbt:115]: Examples should not have a Jacoco report. We are even skipping tests in them. They are not a deliverable package but rather a showcase.

Comment (miroslavpojer) [reply build.sbt]: Code coverage measuring removed in commit 96d38265b64c3b092a624e733decebe52c716df9.

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**